### PR TITLE
feat: add ability to pass name to slack channel data source

### PR DESCRIPTION
**Pull request by connorteague**
## Description

Adding the ability to pass the channel name into the data source for slack channels. This will make our automations simpler as we mostly configure the slack channel via name, not id. 

## PR readiness 

- [x] Relevant documentation has been updated if this PR adds or updates attributes, resources, or data sources.
- [x] An entry has been added to the changelog, if necessary.

### CODE REVIEW

**Comment by wilsonehusin** [comment]
Hey @connorteague thanks for submitting! Initially I implemented this without using Slack channel names because name could get changed and wouldn't be time-proof, so I worry that this would unexpectedly break integrations in surprising ways. While internally FH keeps track of channel names, we can't inform Terraform clients that they need to update. What do you think?

I also saw that you've added `Restricted` field on runbooks — we received a separate contribution (https://github.com/firehydrant/terraform-provider-firehydrant/pull/142) a few weeks back and I was planning to get it sometime next week.

**Comment by connorteague** [comment]
Hey @wilsonehusin that's a good point. If the slack channel changes unexpectedly, I would assume the maintainer of the terraform plan would see that update when they go to apply the the terraform plan. From that perspective, I think any adverse behavior would be caught in the planning stages of the terraform run. 

And thanks for catching that.. reverted in my latest push.

**Comment by wilsonehusin** [comment]
> I would assume the maintainer of the terraform plan would see that update when they go to apply the the terraform plan

That's fair — how do you run Terraform today? Some of our customers are running it in CI pipelines and my understanding is that the people who would change Slack channel name may not be the same people responsible to maintain the infrastructure-as-code.

So if we were to support lookup by name, I'd like to put up a heads up warning for anyone taking this route in the documentation. I'll put up my requests in the review.

File: firehydrant/slack_channels.go
Comment (wilsonehusin) [inline firehydrant/slack_channels.go:62]: let's keep this one as "id" and "name" (also below) — "query" has a different meaning and refers to a different parameter in our API.

**Review by wilsonehusin — COMMENTED** [review]
Thanks again for submitting contribution — much appreciated!

**Comment by connorteague** [comment]
> > I would assume the maintainer of the terraform plan would see that update when they go to apply the the terraform plan
> 
> That's fair — how do you run Terraform today? Some of our customers are running it in CI pipelines and my understanding is that the people who would change Slack channel name may not be the same people responsible to maintain the infrastructure-as-code.
> 
> So if we were to support lookup by name, I'd like to put up a heads up warning for anyone taking this route in the documentation. I'll put up my requests in the review.

Good points. I think the changes you are proposing makes sense. I'll get those incorporated this morning. 

We run our changes with CI as well with GitHub Actions that post any changes to the PR when someone is updating any definition.

File: firehydrant/slack_channels.go
Comment (wilsonehusin) [inline firehydrant/slack_channels.go:68]:
```suggestion
				"id":         params.ID,
				"name":       params.Name,
```

File: firehydrant/slack_channels.go
Comment (wilsonehusin) [inline firehydrant/slack_channels.go:78]:
```suggestion
		"id":         channels.Channels[0].ID,
		"name":       channels.Channels[0].Name,
```

**Review by wilsonehusin — COMMENTED** [review]
woops looks like two items were missed, but we should be good to go after this!

**Comment by connorteague** [comment]
Seems like some tests are failing from a lack of an FH_API_KEY environment variable.. any thouhgts? @wilsonehusin

**Comment by wilsonehusin** [comment]
oh this is probably because it's non-org contribution. let me find out what's going on.

**Comment by connorteague** [comment]
Have there been any updates on this yet? @wilsonehusin

**Comment by wilsonehusin** [comment]
ran the tests on my end and looks OK, thank you!

**Review by wilsonehusin — APPROVED** [review]

**Comment by wilsonehusin** [comment]
FYI @connorteague we'll push a tagged release together with #142, hopefully early next week!

**Pull request merged**
